mungegithub/mungers/inactive-review-handler.go

```diff
@@ -0,0 +1,175 @@
+	items := matchers.Items{}.
+		AddComments(comments...).
+		AddReviewComments(reviewComments...)
+
+	if filterHuman {
+		items.Filter(matchers.HumanActor())
+	}
+
+	lastDate := items.LastDate(issueCreatedAt)
+
+	return lastDate
+}
+
+// Determine whether the PR is active based on number of hours
+// A PR is inactive if it is not being modified for more than one week
+func (i *InactiveReviewHandler) isPRActive(issueCreatedAt *time.Time, comments []*githubapi.IssueComment, reviewComments []*githubapi.PullRequestComment) bool {
+	lastDate := i.getLastDate(issueCreatedAt, comments, reviewComments, false)
+
+	return time.Since(*lastDate) <= 7*24*time.Hour
+}
+
+// Suggest a new reviewer who is NOT any of the existing reviewers
+// (1) get all current assignees for the PR
+// (2) get potential owners of the PR using Blunderbuss algorithm (calling getPotentialOwners() function)
+// (3) filter out current assignees from the potential owners
+// (4) if there is no any new reviewer available, the bot will encourage the PR author to ping all existing assignees
+// (5) otherwise, select a new reviewer using Blunderbuss algorithm (calling selectMultipleOwners() function with number of assignees parameter of one)
+// Note: the munger will suggest a new reviewer when the PR currently does not have any reviewer
+func (i *InactiveReviewHandler) suggestNewReviewer(issue *githubapi.Issue, potentialOwners weightMap, weightSum int64) string {
+	var newReviewer string
+
+	if issue.Assignees != nil && len(issue.Assignees) > 0 {
+		for _, oldReviewer := range issue.Assignees {
+			login := *oldReviewer.Login
+
+			for potentialOwner := range potentialOwners {
+				if login == potentialOwner {
+					weightSum -= potentialOwners[login]
+					delete(potentialOwners, login)
+					break
+				}
+			}
+		}
+	}
+
+	if len(potentialOwners) == 0 {
+		return newReviewer
+	}
+
+	newReviewer = selectMultipleOwners(potentialOwners, weightSum, 1)[0]
+
+	return newReviewer
+}
+
+// Munge is the workhorse encouraging PR author to assign a new reviewer
+// after getting no response from current reviewer for one week
+// The algorithm:
+// (1) find last modification time of the reviewer
+// (2) if the date is one week or longer before today's date, create a comment
+//     encouraging the author to assign a new reviewer and unassign the old reviewer
+// (3) suggest the new reviewer using Blunderbuss algorithm, making sure the old reviewer is not suggested
+// Note: the munger will post at most 5 comments
+func (i *InactiveReviewHandler) Munge(obj *github.MungeObject) {
+	issue := obj.Issue
+
+	// do not suggest new reviewer if it is not a PR or the PR has no author information
+	if !obj.IsPR() || issue.User == nil || issue.User.Login == nil {
+		return
+	}
+
+	comments, ok := obj.ListComments()
+	if !ok {
+		return
+	}
+
+	reviewComments, ok := obj.ListReviewComments()
+	if !ok {
+		return
+	}
+
+	if i.isPRActive(obj.Issue.CreatedAt, comments, reviewComments) {
```

Comment:
i lean towards having the active/inactive check in a help function - there are many procedures in the main `Munge` function. furthermore, i am able to test it more thoroughly in this way.